Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+console.log('STEP 5: Update a Contact');
+
+function findContact(name, contactList) {
+  for (const contact of contactList) {
+    if (contact.name === name) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function updateContact(name, newPhone, newEmail, contactList) {
+  const isContactFound = findContact(name, contactList);
+  if (isContactFound) {
+    const contact = contactList.find(contact => contact.name === name);
+    contact.phone = newPhone;
+    contact.email = newEmail;
+    console.log(`Contact updated successfully.`);
+  }else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+
+let contacts5 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+updateContact("Alice", "555-555-5555", "alice_new@gmail.com", contacts5);
+updateContact("Charlie", "111-111-1111", "charlie@gmail.com", contacts5);
+
+/*
+-----------------------------------------------------------
+  STEP 6: Remove a Contact
+-----------------------------------------------------------
+Function: removeContact(name)
+- Finds the index of the contact with 'name' using
+  findIndex() or a loop.
+- Splices it from the array if found.
+- Logs "Contact removed successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+console.log(' STEP 6: Remove a Contact');
+
+let contacts6 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function removeContact(name, contactList) {
+  const contactIndex = contactList.findIndex(contact => contact.name === name);
+
+  if (contactIndex !== -1) {
+    contactList.splice(contactIndex, 1);
+    console.log(`Contact removed successfully.`);
+  }else {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+removeContact(`Alice`, contacts6);
+removeContact(`Charlie`, contacts6);
+
+/*
+-----------------------------------------------------------
+  STEP 7: Testing Our Functions
+-----------------------------------------------------------
+Below are some sample function calls to demonstrate the
+Contact Book in action.
+*/
+console.log('STEP 7: Testing Our Functions');
+
+let contacts7 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' },
+  { name: 'Charlie', phone: '222-555-8888', email: 'charlie@gmail.com' }
+];
+
+
+function addContact(name, phone, email, contactList) {
```

Comment:
I belive you have addContact defined above already, you don't need to define it again here.